xinference/model/embedding/core.py

```diff
@@ -203,14 +224,19 @@ def to(self, *args, **kwargs):
             )
 
     def create_embedding(self, sentences: Union[str, List[str]], **kwargs):
+        from FlagEmbedding import BGEM3FlagModel
         from sentence_transformers import SentenceTransformer
 
         kwargs.setdefault("normalize_embeddings", True)
 
+        if kwargs.get("return_sparse") and "m3" in self._model_spec.model_name.lower():
+            self._kwargs["hybrid_mode"] = True
```

Comment:
This looks a bit disruptive to the design, I don't know if there is a more elegant way.

Reply:
Are you referring only to the if judgment part or the subsequent reload part?

Reply:
I  mean the reload part.

Reply:
How about loading bge-m3 when specifying `hybrid_mode=True`? This can be done in load.